### PR TITLE
Feature: Extended person features and fields

### PR DESCRIPTION
**Pull request by pyrello**
Addresses #1553 

Adds a person_extended config split that adds additional fields for the person content type.

* Adds person type taxonomy and entity reference field.
* Adds logic in sitenow_people that will show/hide fields when the person type field is present. It looks for the presence of person type (lowercased) in the field name.
* Provides a Research Areas field which only shows when the faculty type is selected (`field_faculty_research_areas`).

I tested this against the default site to make sure that the logic related to the person type field gracefully degrades when the split isn't turned on.

This meets the minimum need for me to continue with migrating content on the gradinformatics.prod.drupal.uiowa.edu site and it does so in a way that doesn't require any other site to adopt the changes. If we decide that we want to adopt any of these fields into the base profile, then we can move them from the split into the profile.

### CODE REVIEW

File: config/default/config_split.config_split.person_extended.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+uuid: 41268a21-7cce-49ad-ab5c-192dcb64e597
+langcode: en
+status: true
```
Comment (joewhitsitt) [inline config/default/config_split.config_split.person_extended.yml]: Status should be `false` for default correct? I just synced gis.sites and saw the fields. person type label was visible but blank (no terms) and required.
Screen Shot 2020-05-21 at 8 28 03 AM

**Comment by pyrello** [comment]
@joewhitsitt Currently, you have to manually add the person types. This is something that I expect to revisit.

**Comment by joewhitsitt** [comment]
Now when i try to enable the split through the UI on a standard site, nothing happens in terms of vocab, fields. i am new to the feature split functionality

**Comment by richardbporter** [comment]
I'm guessing you need to import config after activating the split: https://github.com/uiowa/uiowa/blob/master/config/features/README.md

**Comment by pyrello** [comment]
Now I am running into the issue where, if there are no person types defined, then I have nothing to check against for the additional fields that are added, and they will all show up. I think this means that I need to have a common fragment in the field name that I can check against to hide all of them if there is no person type. I made the decision previously to name the "Research areas" field `field_faculty_research_areas` because adding `person` in it would have made it too long. I would be tempted to switch to a naming convention that doesn't include `field_` except that all the fields have already been named that way.

I'm open to suggestions about how to proceed.

**Review by joewhitsitt — APPROVED** [review]
I could see website coming back to the default config soon. I'd like to identify who is supposed to guide us on what a UofI person should look like in terms of fields. Thanks for making changes quickly

**Comment by joewhitsitt** [comment]
Darn, didn't see your comment before i approved...

**Comment by pyrello** [comment]
Adjusted the naming scheme for conditional fields to start with `person_pt_`. This also allows for me to have a more stringent check when checking for a match against a taxonomy term. Since the terms can be managed on site, there was the possibility of someone creating a match for an existing field by accident. This is unlikely now since they would have to include `pt_` in their term.

Also noting that this doesn't have proper handling for taxonomy terms that are multiple words, like "Student worker". I'll create a follow-up issue since this isn't getting deployed universally yet.

I tested against a site that had terms set for person type and a site that did not.

**Review by joewhitsitt — APPROVED** [review]

**Pull request merged**